### PR TITLE
US4.2: Implement API Endpoint for Commenting on Posts with Error Handling

### PR DESCRIPTION
**Pull request by gmarvega**
### **Description:**  
This PR adds API endpoints that allow authenticated users to interact with comments on posts. Users can add, retrieve, and delete comments on posts. The endpoints include robust error handling to address common edge cases and provide clear feedback to the frontend. 

### **Changes Included:**  

- **Add Comment:**
  - Implemented functionality to allow an authenticated user to add a comment to a specific post.
  - Ensure comments are linked to the appropriate post and user.

- **Retrieve Comments:**
  - Implemented functionality to retrieve all comments for a specific post.

- **Delete Comment:**
  - Implemented functionality to allow users to delete their own comments.
  - Return an error if the comment does not exist or if the user is unauthorized to delete the comment.

### **Error Handling:**
- **401 Unauthorized:** For unauthenticated requests attempting to comment on posts.
- **403 Forbidden:** For attempts to delete comments by non-owners.
- **404 Not Found:** For non-existent posts or comments.
- **400 Bad Request:** For invalid input data during comment creation or updates.

### **Tests:**
- **Unit Tests Added:**
  - Validate successful creation, retrieval, and deletion of comments.
  - Error handling for unauthorized actions, unauthenticated requests, and invalid input data.
  - Cover edge cases:
    - Adding comments to non-existent posts.
    - Deleting non-existent comments.
    - Retrieving comments for posts without any comments.

### CODE REVIEW

File: backend/app/api/routes/comments.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+from app import crud, utils
+from app.api.deps import (get_current_user)
+
+router = APIRouter()
+
+# Create comment endpoint
+@router.post("/",
+             dependencies=[Depends(get_current_user)])
+def create_comment(new_comment: Comment, session: Session = Depends(get_session), current_user: User = Depends(get_current_user)):
+    utils.check_ownership(current_usr_id=current_user.id, check_usr_id=new_comment.user_id)
```
Comment (AlejandroDS1) [inline backend/app/api/routes/comments.py:21]: Quizá seria mejor crear una clase CreateComment que no contenga el userID, a nivel de API ya te autenticas con el token, no creo que sea necesario pasar el ID en la petición.

**Review by AlejandroDS1 — APPROVED** [review]
LGTM, he comentado un detalle por si quieres modificarlo para que el body del endpoint se vea un poco mejor.

**Review by AlejandroDS1 — APPROVED** [review]
